src/main/java/org/rhizome/server/domain/article/domain/ArticleReference.java

```diff
@@ -0,0 +1,46 @@
+package org.rhizome.server.domain.article.domain;
+
+import org.rhizome.server.support.BaseTimeEntity;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class ArticleReference extends BaseTimeEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "source_post_id", nullable = false)
+    private Article sourceArticle;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "target_post_id", nullable = false)
+    private Article targetArticle;
+
+    @Builder
+    private ArticleReference(Article sourceArticle, Article targetArticle) {
+        this.sourceArticle = sourceArticle;
+        this.targetArticle = targetArticle;
+    }
+
+    public ArticleReference createArticleReference(Article sourceArticle, Article targetArticle) {
+        return ArticleReference.builder()
+                .sourceArticle(sourceArticle)
+                .targetArticle(targetArticle)
+                .build();
+    }
```

Comment:
해당부분도 정적팩토리메서드로 사용한다면 static이 필요합니다 
```suggestion
    public static ArticleReference create(Article sourceArticle, Article targetArticle) {
        return ArticleReference.builder()
                .sourceArticle(sourceArticle)
                .targetArticle(targetArticle)
                .build();
    }
```
그리고 굳이 createXXX 이렇게 할 필욘없을것같아여

Reply:
나중에 다른 엔티티(객체)들 생길때를 생각하면 이렇게 짓는게 가독성에 좋지 않을까요?

Reply:
어차피 메서드 사용할 때 `ArticleReference.create()`와 같은 형태로 사용해야돼서 기존처럼 작업하면 `ArticleReference.createArticleReference()` 로 사용하게 되는데 그건 다른 객체들이 생길때 가독성에 문제를 가져오진 않을 것 같아요

오히려 Article 객체가 생성되는게 여러가지가 된다면 이름을 변경해볼만한 여지가 있을 것 같은데 현재는 없으니 간단하게 가도 될 것 같아요

Reply:
그렇네요 `ArticleReference.create()` 이 생각을 못햇네요